src/core/Form/Textarea/Textarea.tsx

```diff
@@ -1,52 +1,161 @@
+      'both' | 'vertical' | 'horizontal' | 'none'
+      @default 'vertical' 
+   */
+  resize?: 'both' | 'vertical' | 'horizontal' | 'none';
+  /** Optional text that is shown after labelText. Will be wrapped in parentheses. */
+  optionalText?: string;
+  /** Unique id
+   * @default uuidV4
+   */
+  id?: string;
+  /** Input name */
+  name?: string;
+}
 
-export class Textarea extends Component<TextareaProps> {
+class BaseTextarea extends Component<TextareaProps> {
   render() {
     const {
-      children,
+      id,
+      className,
       disabled = false,
+      children,
+      onClick,
+      labelMode,
+      labelText,
+      hintText,
       status,
-      className,
+      statusText,
+      visualPlaceholder,
+      resize: dismissResize,
+      optionalText,
+      'aria-describedby': ariaDescribedBy,
       ...passProps
-    } = withSuomifiDefaultProps(this.props);
+    } = this.props;
+
+    const onClickProps = !!disabled ? {} : { onMouseDown: onClick };
+    const statusTextId = `${id}-statusText`;
+    const hintTextId = `${id}-hintText`;
+
+    const getDescribedBy = () => {
+      if (statusText || hintText || ariaDescribedBy) {
+        return {
+          'aria-describedby': [
+            ...(statusText ? [statusTextId] : []),
+            ...(hintText ? [hintTextId] : []),
+            ...(ariaDescribedBy ? [ariaDescribedBy] : []),
+          ].join(' '),
+        };
+      }
+      return {};
+    };
+
     return (
-      <StyledTextarea
-        disabled={disabled}
-        className={classnames(className, {
+      <HtmlDiv
+        className={classnames(baseClassName, className, {
           [textareaClassNames.disabled]: !!disabled,
           [textareaClassNames.error]: status === 'error' && !disabled,
         })}
-        status={status}
-        {...passProps}
       >
-        {children}
-      </StyledTextarea>
+        <LabelText
+          htmlFor={id}
+          labelMode={labelMode}
+          as="label"
+          optionalText={optionalText}
+        >
+          {labelText}
+        </LabelText>
+        <HintText id={hintTextId}>{hintText}</HintText>
+        <HtmlDiv className={textareaClassNames.textareaContainer}>
+          <HtmlTextarea
+            id={id}
+            className={textareaClassNames.textarea}
+            disabled={disabled}
+            defaultValue={children}
+            placeholder={visualPlaceholder}
+            {...{ 'aria-invalid': status === 'error' }}
+            {...getDescribedBy()}
+            {...passProps}
+            {...onClickProps}
+          />
+        </HtmlDiv>
+        <StatusText id={statusTextId} status={status}>
+          {statusText}
+        </StatusText>
+      </HtmlDiv>
     );
   }
 }
+
+const StyledTextarea = styled(
+  ({
+    tokens,
+    id: propId,
+    ...passProps
+  }: TextareaProps & InternalTokensProp) => (
+    <AutoId id={propId}>
+      {(id) => <BaseTextarea id={id} {...passProps} />}
+    </AutoId>
```

Comment:
Added **AutoId** here to be consistent as it is added here with _Checkbox_ also.